bin/run.js

```diff
@@ -28,67 +28,76 @@
  * ```
  */
 
-const chalk = require('chalk');
-const yaml = require('js-yaml');
-const fs = require('fs');
-const path = require('path');
-const { spawn } = require('child_process');
-const workspace = process.cwd();
-const { runUses } = require('../lib/uses');
-
-const gitChangedFiles = require('actions-utils/git-changed-files');
-const getBaseRef = require('actions-utils/get-base-ref');
+const chalk = require("chalk");
+const yaml = require("js-yaml");
+const fs = require("fs");
+const path = require("path");
+const { spawn } = require("child_process");
+
+const { runUses } = require("../lib/uses");
+
+const gitChangedFiles = require("actions-utils/git-changed-files");
+const getBaseRef = require("actions-utils/get-base-ref");
+
+// Walk up file system until we find the "workspace root", which must have a `.github/workflow-templates` directory.
+let workspace = process.cwd();
+while (!fs.existsSync(path.join(workspace, ".github/workflow-templates"))) {
+  if (workspace === "/") {
+    throw new Error(
+      `Unable to determine "workspace" -- no '.github/workflow-templates' directory found in parent directories of ${process.cwd()}`
+    );
+  }
+  workspace = path.dirname(workspace);
+}
 
 let _verbose = false;
 
 const debug = (...args) => {
-    if (_verbose) {
-        console.log(...args);
-    }
+  if (_verbose) {
+    console.log(...args);
+  }
 };
 
 const plural = (num, single, plural) => (num === 1 ? single : plural);
 
 const matches = (jobId /* :string*/, type) => {
-    return (
-        jobId === type ||
-        jobId.endsWith('-' + type) ||
-        jobId.endsWith('_' + type)
-    );
+  return (
+    jobId === type || jobId.endsWith("-" + type) || jobId.endsWith("_" + type)
+  );
 };
 
 function escapeRegExp(string /*: string*/) {
-    return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
```

Comment:
I wonder why your linter is going from `'` → `"`. It's now making me realize that I have seen some `"` → `'` → `"`s lately, specifically in files in `mobile/.github/workflow-templates`, so I wonder if we're all getting some `eslintrc` collisions somewhere. Not blocking for this, but maybe worth checking out which folders/repos are pointing to which rcs.